.github/workflows/black.yml

```diff
@@ -0,0 +1,13 @@
+name: Lint
+
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: psf/black@stable
+        with:
+          options: "--check --diff --verbose"
+          version: "23.1a1"
```

Comment:
You can actually directly use pre-commit as a job, so you don't have to re-specify things. Something like this should work:

```yaml
jobs:
  pre-commit:
    name: pre-commit
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-python@v3
      - uses: pre-commit/action@v3.0.0
```

Reply:
I actually saw this action. (https://github.com/pre-commit/action)
But did not use it since there was a depreciation notice.
Will change it to directly use pre-commit action.